infra/README.md

```diff
@@ -0,0 +1,31 @@
+This is a work in progress to use Terraform to manage our cloud
+infrastructure.
+
+Instructions:
+
+- You will need a GCP project.  Create a service account for Terraform
```

Comment:
Might be worth mentioning the required APIs that should be enabled, otherwise it'll take a few Terraform runs. E.g. https://console.developers.google.com/marketplace/details/google-cloud-platform/compute-engine, https://console.developers.google.com/apis/library/cloudresourcemanager.googleapis.com.

Reply:
Did you have to enable servicenetworking?  I had a resource for that:

```
resource "google_project_service" "service_networking" {
  service = "servicenetworking.googleapis.com"
}
```

and I'm wondering if it didn't work.

Reply:
Yes, here's the error message I got:

```
google_project_service.service_networking: Creating...
google_compute_global_address.gateway: Creation complete after 15s [id=projects/<redacted>/global/addresses/gateway]
kubernetes_secret.global_config: Modifying... [id=default/global-config]
kubernetes_secret.global_config: Modifications complete after 1s [id=default/global-config]
google_project_service.service_networking: Still creating... [10s elapsed]
google_project_service.service_networking: Still creating... [20s elapsed]
google_project_service.service_networking: Creation complete after 28s [id=<redacted>/servicenetworking.googleapis.com]

Error: googleapi: Error 403: Service Networking API has not been used in project <redacted> before or it is disabled. Enable it by visiting https://console.developers.google.com/apis/api/servicenetworking.googleapis.com/overview?project=<redacted> then retry. If you enabled this API recently, wait a few minutes for the action to propagate to our systems and retry., accessNotConfigured
```

After enabling the API, I still got a 403, but that was because of missing permissions for the service account:

```
google_service_networking_connection.private_vpc_connection: Creating...

Error: googleapi: Error 403: The caller does not have permission, forbidden
```

Reply:
Ah, looks like it needs an explicit dependency.  That's annoying and race condition-y.  I'll just put the gcloud enable in the README.  Thanks!

Reply:
https://github.com/terraform-google-modules/terraform-google-project-factory/tree/master/modules/project_services#project-api-activation might be worth a try.

Reply:
I looked into this.  It just expands into a set of "google_project_service" resources like I had originally, but doesn't appear to address the dependency issue.